Upstream-Status: Inappropriate [backport from upstream]

In May, people already reported the mesa-dri build issue when upgrading to
glproto-1.4.13 and dri2proto-2.4:
http://lists.freedesktop.org/archives/mesa-dev/2011-May/007331.html

The story is: Jesse tried to change the definitions of some structs
in glproto, dri2proto and mesa-dri, but the change in mesa-dri side was accepted
and then reverted, see
http://cgit.freedesktop.org/mesa/mesa/commit/?id=dc4a3edcf9d414d5098618d5d3b79d8dd90adcc3

so in glproto and driproto's sides, we have to revert the change, too:
http://cgit.freedesktop.org/xorg/proto/dri2proto/commit/?id=f3c211e1ae183dab5c7529814c9f42af2c29fc6c

We could remove this patch when upgrading to newer glproto, dri2proto and
mesa-dri in future.

Wed Jun 15 14:33:14 CST 2011
Dexuan Cui <dexuan.cui@intel.com>

-------------------------------------------------------

commit f3c211e1ae183dab5c7529814c9f42af2c29fc6c
Author: Jesse Barnes <jbarnes@virtuousgeek.org>
Date:   Thu May 5 09:22:39 2011 -0700

    Revert "dri2proto: make DRI2 swap event match GLX spec"
    
    This reverts commit 0ca3778de195a82087d0f07415a1cf8fc94f5b0a.
    
    This broke compatibility by renaming existing struct fields.  So revert
    in favor of a new struct.

diff --git a/dri2proto.h b/dri2proto.h
index ff76355..9708a4a 100644
--- a/dri2proto.h
+++ b/dri2proto.h
@@ -290,13 +290,13 @@ typedef struct {
     CARD8 pad;
     CARD16 sequenceNumber B16;
     CARD16 event_type B16;
-    CARD16 pad2;
     CARD32 drawable B32;
     CARD32 ust_hi B32;
     CARD32 ust_lo B32;
     CARD32 msc_hi B32;
     CARD32 msc_lo B32;
-    CARD32 sbc B32;
+    CARD32 sbc_hi B32;
+    CARD32 sbc_lo B32;
 } xDRI2BufferSwapComplete;
 #define sz_xDRI2BufferSwapComplete 32
 
